src/background/services/paymentSession.ts

```diff
@@ -336,4 +274,57 @@ export class PaymentSession {
     this.amount = amount.toString()
     this.intervalInMs = Number((amount * BigInt(HOUR_MS)) / BigInt(this.rate))
   }
+
+  private async payContinuous() {
+    try {
+      const outgoingPayment =
+        await this.openPaymentsService.createOutgoingPayment({
+          walletAddress: this.sender,
+          incomingPaymentId: this.incomingPaymentUrl,
+          amount: this.amount
+        })
+      const { receiveAmount, receiver: incomingPayment } = outgoingPayment
+      const monetizationEventDetails: MonetizationEventDetails = {
+        amountSent: {
+          currency: receiveAmount.assetCode,
+          value: transformBalance(receiveAmount.value, receiveAmount.assetScale)
+        },
+        incomingPayment,
+        paymentPointer: this.receiver.id
+      }
+
+      sendMonetizationEvent({
+        tabId: this.tabId,
+        frameId: this.frameId,
+        payload: {
+          requestId: this.requestId,
+          details: monetizationEventDetails
+        }
+      })
+
+      // TO DO: find a better source of truth for deciding if overpaying is applicable
+      if (this.intervalInMs > 1000) {
+        this.tabState.saveOverpaying(this.tab, this.url, {
+          walletAddressId: this.receiver.id,
+          monetizationEvent: monetizationEventDetails,
+          intervalInMs: this.intervalInMs
+        })
+      }
+    } catch (e) {
+      if (isKeyRevokedError(e)) {
+        this.events.emit('open_payments.key_revoked')
+      } else if (isTokenExpiredError(e)) {
+        this.openPaymentsService.rotateToken()
+      } else if (e instanceof OpenPaymentsClientError) {
+        // We need better error handling.
+        if (e.status === 400) {
+          await this.setIncomingPaymentUrl()
+        }
+
+        // TODO: Check what Rafiki returns when there is no amount
+        // left in the grant.
+        throw new Error(e.message)
+      }
```

Comment:
In case of errors, I wonder if there should be a way to pause other payments momentarily, so they don't trigger same errors over and over until they're handled? What do you think @raducristianpopa? Or is that fine as is?

Reply:
pausing them would reintroduce the difference of frequency

Reply:
I'm fine with the implementation as is for now. We can revisit this in the future.